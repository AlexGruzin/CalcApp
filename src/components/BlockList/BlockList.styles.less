
.block-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.block-list__item {
  display: flex;
  align-items: center;
  justify-content: center;
  background: var(--grey100);
  border-radius: 13px;
  width: 161px;
  height: 72px;
  margin: 7px 14px 7px 0;
  cursor: pointer;
}

.block-list__item-inner {
  display: flex;
  height: 20px;
}
