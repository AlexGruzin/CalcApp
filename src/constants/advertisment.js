export const ADVERT = {
  CORDIAL: 'CORDIAL',
  FACEBOOK: 'FACEBOOK',
  GOOGLE_ADS: 'GOOGLE_ADS',
  GOOGlE_ANALYTICS: 'GOOGLE_ANALYTICS',
  INSTAGRAM: 'INSTAGRAM',
  KLAVIYO: 'KLAVIYO',
  TIKTOK: 'TIKTOK',
  TWITTER: 'TWITTER',
  SHOPIFY: 'SHOPIFY',
  SHOPIFY_SIGNUP: 'SHOPIFY_SIGNUP',
};

export const SOCIAL = {
  ENVELOPE: 'ENVELOPE',
  BITMAP: 'BITMAP',
};

export const ADVERT_ICONS = {
  [ADVERT.FACEBOOK]: 'facebook.png',
  [ADVERT.INSTAGRAM]: 'instagram.png',
  [SOCIAL.ENVELOPE]: 'envelope.png',
  [SOCIAL.BITMAP]: 'bitmap.png',
};
