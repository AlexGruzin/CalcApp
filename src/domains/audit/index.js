import AllActionsTypes from './actionTypes';
import * as AllActionsCreators from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export { AllActionsTypes as ActionsTypes, AllActionsCreators as ActionsCreators, reducer, selectors };
