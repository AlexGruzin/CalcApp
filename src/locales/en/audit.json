{
  "title": "Audit",
  "letsGo": "Let's go!",
  "manyStores": "Many stores accidentally run ads for products that are out of stock! This frustrates many shoppers and sales are lost. The NogBots can automagically make sure that shoppers will only see ads for items that they can and will want to buy.",
  "submit": "Submit",
  "connectYourStore": "Connect your store catalogue and channels to create your marketing plan. We even encourage tracking services like Google Analytics.",
  "connected": "Connected",
  "seeAllYourMarketing": "See all your marketing in",
  "onePlace": "one place.",
  "threeEasySteps": "Three Easy Steps to Get Started With Calc",
  "auditItem": "Audit We look at your store, analytics, and advertising history to build your marketing audit",
  "forecastItem": "Forecast We present you with achievable goals designed to your business’s strength. You get to tailor the plan.",
  "curateItem": "Curate Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
  "deacadesOfExperience": "Deacades of Experience",
  "workingWithClients": "Working with clients in ecommerce for over 20 years, we know digital marketing.",
  "handleTheGuesswork": "We’ll Handle the Guesswork",
  "ourCalcBots": "Our CalcBots are designed to look at transactions in real time - we know which marketing mix is doing the best.",
  "proTip": "ProTip!",
  "storeCatalogue": "Store Catalogue",
  "analytics": "Analytics",
  "required": "Required",
  "optional": "Optional",
  "save": "Save",
  "advertisingChannels": "Advertising Channels",
  "runMyAudit": "Run my Audit",
  "connectionLoad": {
    "weWorkingHard": "We are working hard to create your Marketing Audit. We will help you identify real attainable goals for your business, as well as the steps to achieve them.",
    "evenThoughAudits": "Even though audits can sometimes take a while to create, the wait is worth it! Feel free to check back on this page, but we’ll let you know when it’s ready.",
    "forNow": "For now, we suggest looking at the ",
    "toBegin": " to begin your marketing plan."
  },
  "forecast": "Forecast page",
  "back": "Back",
  "next": "Next"
}
