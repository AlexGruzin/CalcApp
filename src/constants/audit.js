import { ADVERT } from 'constants/advertisment';

export const AUDIT_STORE_COMPLETE = 'STORE';
export const AUDIT_ANALYTICS_COMPLETE = 'ANALYTICS';
export const AUDIT_CHANNELS_COMPLETE = 'CHANNELS';

export const ADVERTISING_CHANNELS = [
  // {
  //   name: ADVERT.CORDIAL,
  //   image: 'cordial-logo.png',
  // },
  {
    name: ADVERT.GOOGLE_ADS,
    image: 'g-ads-logo.png',
  },
  // {
  //   name: ADVERT.FACEBOOK,
  //   image: 'facebook-logo.png',
  // },
  // {
  //   name: ADVERT.INSTAGRAM,
  //   image: 'instagram-logo.png',
  // },
  // {
  //   name: ADVERT.KLAVIYO,
  //   image: 'klaviyo-logo.png',
  // },
  // {
  //   name: ADVERT.TIKTOK,
  //   image: 'tiktok-logo.png',
  // },
  // {
  //   name: ADVERT.TWITTER,
  //   image: 'twitter-logo.png',
  // },
];

export const USER_STATUS = {
  FIRST_TIME: 'FIRST_TIME',
  AUDIT_PASSED: 'AUDIT_PASSED',
};
