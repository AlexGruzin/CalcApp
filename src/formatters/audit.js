import { REVENUE_TYPES } from 'constants/auditDashboard';
import orderBy from 'lodash/orderBy';

export const formatAuditStep = data => {
  const { user_organization_role, wizard_step } = data;

  return {
    userOrganizationRole: user_organization_role,
    wizardStep: wizard_step,
  };
};

export const formatChannels = data => {
  if (!data) return {};

  return {
    [REVENUE_TYPES.EMAIL]: data.Email,
    [REVENUE_TYPES.REFERRAL]: data.Referral,
    // [REVENUE_TYPES.SEO]: data.SEO,
    [REVENUE_TYPES.PAID_SEARCH]: data['Paid Search'],
    [REVENUE_TYPES.AFFILIATES]: data.Affiliates,
    [REVENUE_TYPES.DIRECT]: data.Direct,
    [REVENUE_TYPES.ORGANIC_SEARCH]: data['Organic Search'],
    [REVENUE_TYPES.DISPLAY]: data.Display,
    [REVENUE_TYPES.SMS]: data.SMS,
    [REVENUE_TYPES.RETARGETING]: data.Retargeting,
    [REVENUE_TYPES.SOCIAL]: data.Social,
    [REVENUE_TYPES.OTHER]: data['(Other)'],
  };
};

export const formatCities = data => {
  if (!data) return [];
  const formatted = Object.keys(data).map(key => ({
    city: key,
    value: data[key],
  }));
  return orderBy(formatted, 'value', 'desc');
};

export const formatKeywords = data => {
  if (!data) return [];
  const formed = Object.keys(data).map(key => ({
    name: key,
    adSpend: data[key].ad_spend,
    impressions: data[key].impressions,
  }));

  return orderBy(formed, 'impressions', 'desc');
};

export const formatAuditDashboardData = data => {
  const { last_updated_date, report_data, store_id } = data;
  const {
    last_updated_date: lastUpdatedDate,
    ad_spend_by_channel,
    avg_cpc,
    avg_keyword_score,
    goals,
    keyword_efficacy,
    last_year_ad_spend,
    last_year_revenue,
    revenue_by_channel: revenueByChannel,
    revenue_by_city,
  } = report_data;

  return {
    lastUpdatedDate: last_updated_date,
    storeId: store_id,
    reportData: {
      lastUpdatedDate,
      adSpendByChannel: {
        ...formatChannels(ad_spend_by_channel),
        // paidSearch: ad_spend_by_channel['Paid Search'], // old
        // display: ad_spend_by_channel['Display'],
      },
      avgCpc: avg_cpc,
      avgKeywordsCore: avg_keyword_score,
      goals, // all examples are missing this field
      keywordEfficacy: formatKeywords(keyword_efficacy),
      lastYearAdSpend: last_year_ad_spend,
      lastYearRevenue: last_year_revenue,
      revenueByChannel: {
        ...formatChannels(revenueByChannel),
      },
      revenueByCity: formatCities(revenue_by_city),
    },
  };
};
