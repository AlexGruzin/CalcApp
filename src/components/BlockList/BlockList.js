import React from 'react';
import PropTypes from 'prop-types';
import { AUDIT_CHANNELS_COMPLETE } from 'constants/audit';

import classNames from 'classnames';

import './BlockList.styles.less';

export const BlockList = React.memo(({ items, className, handleClick, step }) => (
  <div className={classNames('block-list', className)}>
    {items.length &&
      items.map(item => (
        <div
          role="button"
          onClick={() => handleClick(item)}
          key={item.name}
          tabIndex={0}
          className={classNames('block-list__item', item.name)}
        >
          <div className={classNames('block-list__indicator', { isActive: step === AUDIT_CHANNELS_COMPLETE })}>
            Connected
          </div>
          <div>
            <img className="block-list__item-inner" src={`/images/${item.image}`} alt="Shopify logo" />
          </div>
        </div>
      ))}
  </div>
));

BlockList.propTypes = {
  items: PropTypes.array,
  className: PropTypes.string,
  handleClick: PropTypes.func,
  step: PropTypes.string.isRequired,
};

BlockList.defaultProps = {
  items: [],
  className: '',
  handleClick: () => undefined,
};

export default BlockList;
