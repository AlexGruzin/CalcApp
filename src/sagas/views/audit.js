import { put, take, race, delay, all } from 'redux-saga/effects';
import { getSurveyData } from 'sagas/actions/survey';
import { push } from 'connected-react-router';
import * as Routing from 'constants/routing';
import SurveyActionTypes from 'sagas/actionTypes/survey';
import ActionTypes from 'domains/audit/actionTypes';
import { getAuditStatus, getAuditSteps } from 'domains/audit/actions';

/**
 * Saga that allows us to make safe request.
 */
function* initAuditSaga() {
  yield put(getAuditSteps());

  if (localStorage.getItem('access_token') !== '' && localStorage.getItem('import_data') === 'true') {
    while (true) {
      yield delay(5000);
      yield put(getAuditStatus());

      yield race({
        success: take(ActionTypes.GET_AUDIT_STATUS_SUCCESS),
        error: take(ActionTypes.GET_AUDIT_STATUS_FAILURE),
      });
    }
  }

  yield put(getSurveyData());

  const { success } = yield race({
    success: take(SurveyActionTypes.GET_SURVEY_DATA_SUCCESS),
    error: take(SurveyActionTypes.GET_SURVEY_DATA_FAILURE),
  });
  let isSurveyPassed = false;

  if (success) {
    // TODO discuss to swap with doNotShowNoMore field
    const { brand_name, industry } = success.payload;
    if (brand_name && industry) isSurveyPassed = true;
  }

  yield take(ActionTypes.GET_AUDIT_RUN_SUCCESS); // wait for audit success finished

  if (isSurveyPassed) {
    yield put(push(Routing.AUDIT_DASHBOARD));
  } else {
    yield put(push(Routing.SURVEY));
  }
}

export default function* auditPageSaga() {
  yield all([initAuditSaga()]);
}
