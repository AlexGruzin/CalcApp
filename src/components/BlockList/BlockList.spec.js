import React from 'react';
import { waitFor, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ADVERTISING_CHANNELS } from 'constants/audit';

import BlockList from './BlockList';

describe('BlockList tests', () => {
  it('Check snapshots of BlockList are equal.', () => {
    const { asFragment } = render(
      <BlockList items={ADVERTISING_CHANNELS} title="title" handleClick={() => undefined} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('BlockList do call handleClick on item click.', async () => {
    const mockOnClick = jest.fn();
    const { getAllByRole } = render(
      <BlockList items={ADVERTISING_CHANNELS} className="name" handleClick={mockOnClick} />
    );

    const btn = getAllByRole('button');
    userEvent.click(btn[0]);

    await waitFor(() => {
      expect(mockOnClick).toBeCalledTimes(1);
    });
  });
});
