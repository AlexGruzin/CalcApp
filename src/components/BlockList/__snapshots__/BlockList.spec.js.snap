// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockList tests Check snapshots of BlockList are equal. 1`] = `
<DocumentFragment>
  <div
    class="block-list"
  >
    <div
      class="block-list__item GOOGLE_ADS"
      role="button"
      tabindex="0"
    >
      <div
        class="block-list__indicator"
      >
        Connected
      </div>
      <div>
        <img
          alt="Shopify logo"
          class="block-list__item-inner"
          src="/images/g-ads-logo.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
